[v1.1][T09-B4] NUSCouples

chenxing1992 opened the pull request:
[V1.0]
- **Updated DG** (User Profile, User Stories, Value Proposition, Use Case, Glossary)
- **Enhanced UI** (Added colour to tag) - Marlene 
- **Enhanced Storage** (Save file in offline mode) - Chen Xing 
- **Enhanced Model** (Deleting Tag) - Samuel 
- **Enhanced Logic** (Added a shorthand equivalent alias for each of the individual commands) - Daniel

[V1.1]
-  **Updated User Guide, About Us page and Readme** (Team portion)
- **Added in Feature contribution in DG by 20/3/2018 23:59** (Team portion) 
- **Successfully communicate with Google Calender API** - Chen Xing
- **Implemented Journal Window** - Samuel
- **Data Encryption** - Marlene

[V1.2]

-  **Implemented Major/Minor Enhancement documentation in DG (Chen Xing, Marlene, Samuel)**  
-  **Implemented one or more Major/Minor explicitly in DG (Team)**
-  **Perform Merge Major/Minor (Chen Xing, Samuel, Marlene)**

[V1.3]

- **Implement TimeTable API from NUSMODS and make changes to test cases**- Marlene
- **Implement Journal that can read and save**- Samuel
- **Successfully Authenticate with Google API** -ChenXing 

[V1.4] 

- **Implement ReadOnlyPerson Interface and create Calendar UI** - Chen Xing


@marlenekoh  @traceurgan @HEARTOFAL1ON

jeffryhartanto commented on README.adoc:
```diff
@@ -1,7 +1,7 @@
 ifdef::env-github,env-browser[:relfileprefix: docs/]
 
-https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
+https://travis-ci.org/CS2103JAN2018-T09-B4/main[image:https://travis-ci.org/CS2103JAN2018-T09-B4/main.svg?branch=master[Build Status]]
```
Good job for updating this badge, later you need to update the other badges.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 * is reasonably comfortable using CLI apps
 
 *Value proposition*: manage contacts faster than a typical mouse/GUI driven app
+*Our team visions to help new students to not get lost in their first day of school or when it rains. Because right now, there is no proper channel that helps them to solve this problem.
```
by "students" you mean "SoC students", right? Try to be as specific as possible.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+* - Campus direction shortcuts for new students and when it rains
+* - Couples having diet issues
+* - Location on where to eat
 * has a need to manage a significant number of contacts
```
If it's only for couples then why do they need this point? One person can have more than partners?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+* - Journal diary on the meeting
+* - Campus direction shortcuts for new students and when it rains
+* - Couples having diet issues
+* - Location on where to eat
```
I think most of these points are features, not target user profile.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 
 *Value proposition*: manage contacts faster than a typical mouse/GUI driven app
+*Our team visions to help new students to not get lost in their first day of school or when it rains. Because right now, there is no proper channel that helps them to solve this problem.
+*Our team visions to help new students to help them find a place to eat because they don’t know the nearest food centre they can eat at.
```
If it's for SoC student, then the answer should always be The Terrace?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -781,14 +781,25 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+*Our team visions to help new students to help them find a place to eat because they don’t know the nearest food centre they can eat at.
+*Our team visions to help students to find common breaks so they can form a meetup session easily
+*Our team visions to help students to jote down journals that happen on the meetup session so that everything is well stored together
+*Our team visions to help NUS couple to solve problem that they faced out of all the hectic workload in NUS which always leads to a disastrous break up.
```
I think this point is more like a conclusion or very similar to "find common breaks". So, may be it can be removed.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,11 +815,29 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* * *` |user |delete a person |remove entries that I no longer need
 
+|`* * *` |As a couple (student) user |Add/Delete/Compare timetable with my partner |keep track of my partner's free time
```
I think share or sync might be better. Using `Add/Delete/Compare` is unclear.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,11 +815,29 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |As a couple (student) user |create a Journal to Add/View/Delete |keep track on life couple event
+
+|`* * *` |As a new student user |search for campus shortcuts |avoid getting lost or drenched from rain
```
If you only aim SoC student, then you need to mention it.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,11 +815,29 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |As a new student user |search for campus shortcuts |avoid getting lost or drenched from rain
+
+|`* * *` |As a couple (student) user | |I can calculate my diet of how many calories i have burnt
```
There is a missing part here.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,11 +815,29 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*`| As a couple (student) user |receive birthday notifications |remember my partner's birthday
+
+|`*`| As a couple (student) user |view my calendar |take note of upcoming events
```
Is calendar different than timetable?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,11 +815,29 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*`| As a couple (student) user |view my calendar |take note of upcoming events
+
+|`*`| As a couple (student) user |I can view my journal |
```
One star means they might not able to view their journal in the early version. Are you sure?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,6 +873,47 @@ Use case ends.
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
```
Should change AddressBook to your app's name.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,6 +873,47 @@ Use case ends.
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User request to add a shortened NUSMODS url to a specific person in the list.
+4.  AddressBook adds the url to the person in the list.
```
So, where is the timetable? Don't assume that the reader knows about NUSMODS.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -844,6 +873,47 @@ Use case ends.
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Add Timetable to Person
```
Later you can add more use cases.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -852,11 +922,17 @@ _{More to be added}_
+.  Should be usable by a novice after reading the <<UserGuide#, User Guide>>.
+.  Should be able to handle any invalid inputs.
+.  Commands used should be intuitive and comprehensible.
+.  Should respond within 2 seconds.
```
Saying, `respond to user inputs` is clearer, I guess.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -852,11 +922,17 @@ _{More to be added}_
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should be usable by a novice after reading the <<UserGuide#, User Guide>>.
+.  Should be able to handle any invalid inputs.
+.  Commands used should be intuitive and comprehensible.
```
`intuitive and comprehensible` is unclear.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -852,11 +922,17 @@ _{More to be added}_
 
 [appendix]
 == Glossary
+* Product survey, pros - able to gather feedback from couples from SOC for better enhancement cons- to get real consistent feedback as couples behave differently and expose differently
```
I think these should be placed in the next section.

jeffryhartanto commented:
@chenxing1992 @marlenekoh @HEARTOFAL1ON @traceurgan  Some comments added..

jeffryhartanto commented on README.adoc:
```diff
@@ -1,10 +1,10 @@
+https://travis-ci.org/CS2103JAN2018-T09-B4/main[image:https://travis-ci.org/CS2103JAN2018-T09-B4/main.svg?branch=master[Build Status]]
+https://ci.appveyor.com/project/marlenekoh/main[image:https://ci.appveyor.com/api/projects/status/qnuriba83lw19gln?svg=true[Build status]]
+https://coveralls.io/github/CS2103JAN2018-T09-B4/main?branch=master[image:https://coveralls.io/repos/github/CS2103JAN2018-T09-B4/main/badge.svg?branch=master[Coverage Status]]
+https://app.codacy.com/app/marlenekoh/main[image:https://api.codacy.com/project/badge/Grade/83f1bb05107a4a1c9a76bbb8d65ef6bb[Codacy Badge]]
```
Good job for updating all badges.

jeffryhartanto commented on README.adoc:
```diff
@@ -15,19 +15,17 @@ ifndef::env-github[]
+* This is a desktop application for Couples in National University of Singapore(NUS).
+* It has a GUI but most of the user interactions happen using a CLI (Command Line Interface).
+* Our Features:
+** Compare NUS timetables and find your partner's location based on his/her timetable,
```
Good... I think it's getting better.

jeffryhartanto commented on docs/AboutUs.adoc:
```diff
@@ -3,53 +3,47 @@
+{empty}[https://github.com/marlenekoh[github]]
 
-Role: Team Lead +
+Role: Developer +
```
I think someone need to be the team leader.

HEARTOFAL1ON replied:
I'll rise up and take the role of team leader if everyone else is agreeable.

For my leadership style, after much contemplation, I am deciding to go for the servitude leadership style. This means that as a leader, I do not only have to be COMPLETELY accountable to my entire team for my actions, to decide on a path to take etc etc..., BUT in the hierarchy structure of the group, the LEADER is at the BOTTOM while all the other MEMBERS are at the TOP above the LEADER.

Many organizations such the international Toastmasters organization also uses a committee structure whereby it is said members are higher in ranking than elected committee members!

I STRONGLY feel it would be a PHENOMENAL success if we not only MAKE IT as a team just before week 13's DEMO, but we COME OUT AT THE VERY TOP!! :+1:

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,7 +10,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-T09-B4/main
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Update this part.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -780,15 +780,28 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+* needs to remember special moments
+* needs to remember special dates
+* needs to schedule meetings
+* needs to find a good time to meet
```
This point is very related to the previous point, isn't it? If so, you can combine them..

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -780,15 +780,28 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+* is currently a student of NUS
+* is in a relationship with a student of NUS
+* needs to remember special moments
+* needs to remember special dates
```
May be you can merge this point to the previous point, since they're similar.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,29 +817,51 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |new user |have commands that are intuitive and easy to remember |
+
+|`* * *` |user who perfers typing over using the mouse |be able to control almost everything from the CLI |
```
These two points should be in NFR. Why?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,29 +817,51 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
 
 |`*` |user with many persons in the address book |sort persons by name |locate a person easily
```
Will there be many persons? If not, then you need to remove unused/unrelated user stories, also in other sections.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,29 +817,51 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |pair of NUS students in a relationship |view my partner's NUS timetable for the week |keep track of my partner's free time
+
+|`* * *` |pair of NUS students in a relationship |replace my partner's NUS timetable |see the most updated timetable
```
Formatting issue around here.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,29 +817,51 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user who values their privacy |encrypt my NUSCouples save data |prevent strangers from reading personal information such as my journal entries
 
 |`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
```
Only one 2 stars? May be you have too many 3 stars?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -804,29 +817,51 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
+1.  User requests to list persons.
+2.  NUSCouples shows a list of persons.
```
I believe you won't need "list" command, right? If so, change this accordingly.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,7 +875,82 @@ Use case ends.
+2.  NUSCouples shows a list of persons.
+3.  User request to add a shortened http://nusmods.com[NUSMODS] timetable url to a specific person in the list.
+4.  NUSCouples adds the url to the person in the list.
+5.  NUSCouples displays the timetable.
```
No request from the user? may be you can merge this to the previous step. Usually, the steps in MSS show interaction between user and the system. So, having two consecutive user/system steps are not recommended..

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,7 +875,82 @@ Use case ends.
+*MSS*
+
+1.  User requests to list persons.
+2.  NUSCouples shows a list of persons.
```
Do you need to list persons?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -852,12 +962,15 @@ _{More to be added}_
+.  Should respond to user inputs within 2 seconds.
+.  Should be able to work on both 32-bit and 64-bit environments.
 
 _{More to be added}_
```
Don't forget to remove this, once you decide not to add any new point.

jeffryhartanto commented on docs/UserGuide.adoc:
```diff
@@ -11,7 +11,7 @@ ifdef::env-github[]
-:repoURL: https://github.com/se-edu/addressbook-level4
+:repoURL: https://github.com/CS2103JAN2018-T09-B4/main
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
```
Change this part.

jeffryhartanto commented on docs/UserGuide.adoc:
```diff
@@ -101,7 +106,8 @@ Edits the name of the 2nd person to be `Betsy Crower` and clears all existing ta
 Finds persons whose names contain any of the given keywords. +
-Format: `find KEYWORD [MORE_KEYWORDS]`
+Format: `find KEYWORD [MORE_KEYWORDS]` +
+Alias: `f KEYWORD [MORE_KEYWORDS]`
```
Do you still need "find" command?

jeffryhartanto commented on docs/UserGuide.adoc:
```diff
@@ -161,18 +169,86 @@ Selects the 1st person in the results of the `find` command.
+Shows the location (by building) of the entered classroom name. +
+Format: `location` +
+Alias: `loc`
+
```
Ensure that these features are present in the user stories.

jeffryhartanto reviewed (commented):
@chenxing1992 @marlenekoh @HEARTOFAL1ON @traceurgan Some comments added. Good job for passing V1.1 milestone. BTW, you should remove/update those things that are unrelated/unneeded in your app.

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -10,9 +10,9 @@ ifdef::env-github[]
+:repoURL: https://github.com/CS2103JAN2018-T09-B4/main
 
-By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
+By: `Team T09-B4`      Since: `Feb 2018`      Licence: `NUS School Of Computing `
```
Don't change the licence..

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -385,6 +385,137 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+
+The sequence diagram below shows interactions within the `Logic` Component for Outh 2.0 endpoints:
+
+image::viewCalendarSeqDiagram.png[width="850"]
```
Why there are some red colored texts and arrow?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -385,6 +385,137 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+The journal feature is implemented by a `Journal`, which resides inside `ModelManager`. It allows the user to create and save journal entries.
+
+//insert class diagram here
+//image::LogicCommandClassDiagram.png[width="800"]
```
Commented out? Why?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -385,6 +385,137 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+//insert class diagram here
+//image::LogicCommandClassDiagram.png[width="800"]
+
+As you can see from the diagram,
```
Where is the diagram?

jeffryhartanto commented on docs/DeveloperGuide.adoc:
```diff
@@ -385,6 +385,137 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+
+The following sequence diagram shows how the newJournal operation works:
+
+image::JournalWindowUIComponentSequenceDiagram.png[width="800"]
```
I suspect the interaction between JournalWindow and UI is incorrect.

jeffryhartanto commented:
@chenxing1992 @HEARTOFAL1ON @marlenekoh @traceurgan Some comments added for DG's implementation section.

marlenekoh committed:
Merge branch 'master' into timetable

marlenekoh committed:
Merge pull request #139 from traceurgan/ug

Minor update to UG

traceurgan committed:
Made Journal entries update automatically by using set instead of editing data directly

marlenekoh committed:
Create reloadTimetableRequestEvent

traceurgan committed:
Mostly working app
All commands work without index
Only one person is allowed.

traceurgan committed:
CheckStyle

marlenekoh committed:
Add command to show timetable on select and hide timetable on deselect

traceurgan committed:
CS

traceurgan committed:
Minor edit

traceurgan committed:
Removal of unneeded files
Tried to fix some testcases

marlenekoh committed:
Get rid of magic number partner index

marlenekoh committed:
Add comparetimetable command and update gitignore

marlenekoh committed:
Checkstyle fixes

marlenekoh committed:
Replace address book names with NUSCouples in UI

traceurgan committed:
Partial Perge

marlenekoh committed:
Merge branch 'master' into timetable

marlenekoh committed:
Merge pull request #155 from marlenekoh/timetable

Select command and deselect command for timetable

marlenekoh committed:
add code for overlapping timetable

now overlapping timetables will show earlier module slot, followed by what's left of second slot

traceurgan committed:
Merge branch 'baseBranchDevMaster' into removeAddressbook

# Conflicts:
#	src/main/java/seedu/address/commons/events/model/AddressBookChangedEvent.java
#	src/main/java/seedu/address/logic/commands/EditCommand.java
#	src/main/java/seedu/address/logic/parser/AddCommandParser.java
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/ReadOnlyAddressBook.java
#	src/main/java/seedu/address/model/person/Person.java
#	src/main/java/seedu/address/model/person/ReadOnlyPerson.java
#	src/main/java/seedu/address/model/person/UniquePersonList.java
#	src/main/java/seedu/address/model/util/SampleDataUtil.java
#	src/main/java/seedu/address/storage/XmlAdaptedPerson.java
#	src/main/java/seedu/address/storage/XmlAddressBookStorage.java
#	src/main/java/seedu/address/storage/XmlSerializableAddressBook.java
#	src/test/java/seedu/address/commons/util/XmlUtilTest.java
#	src/test/java/seedu/address/model/AddressBookTest.java
#	src/test/java/seedu/address/model/ModelManagerTest.java
#	src/test/java/seedu/address/storage/StorageManagerTest.java
#	src/test/java/seedu/address/storage/XmlSerializableAddressBookTest.java
#	src/test/java/seedu/address/ui/CommandBoxTest.java
#	src/test/java/systemtests/SampleDataTest.java

marlenekoh committed:
Merge branch 'master' into UI

traceurgan committed:
App starts with no person if there is no data

marlenekoh committed:
minor refactor

marlenekoh committed:
Merge remote-tracking branch 'origin/timetable' into timetable

marlenekoh committed:
add compare timetable feature

marlenekoh committed:
Refactoring

marlenekoh committed:
Merge branch 'master' into updateDocs

marlenekoh committed:
update docs

marlenekoh committed:
temp

chenxing1992 committed:
Create CSS for our app

chenxing1992 committed:
bug fix#1

chenxing1992 committed:
asd

traceurgan committed:
Tests

traceurgan committed:
Added marlenes new files

traceurgan committed:
Attempted merge:
Select command doesnt show timetable, shows timetable on mouse press
deselect works, doesnt show error even when no person exists

traceurgan committed:
Fixed Bug. Clear journal doesnt allow you to make a new journal afterwards

traceurgan committed:
Handled null pointer exception
edit, select delete all work as well.

marlenekoh committed:
Bug fixes

Fixed lag in deselect
Fixed bug does not show module code again after comparing then selecting
Fixed bug show timetable on delete

marlenekoh committed:
Fixed checkstyle

marlenekoh committed:
Merge branch 'master' into timetable

traceurgan committed:
Checkstyle

marlenekoh committed:
Merge pull request #167 from marlenekoh/timetable

Timetable bug fixes

marlenekoh committed:
Merge branch 'master' into Implement-CSS-for-UI

marlenekoh committed:
Merge pull request #166 from chenxing1992/Implement-CSS-for-UI

Create CSS for our app to have new look and feel!!

marlenekoh committed:
Merge remote-tracking branch 'CS2103JAN2018-T09-B4/master'

marlenekoh committed:
Refactor and move files

marlenekoh committed:
Merge branch 'timetable' of https://github.com/marlenekoh/addressbook-level4 into timetabletemp

marlenekoh committed:
Merge branch 'timetabletemp' into timetable

marlenekoh committed:
fixed update bug

marlenekoh committed:
Fixed scrolling bug

marlenekoh committed:
ctrl click will deselect and hide timetable

traceurgan committed:
Merge branch 'master' into UI

marlenekoh committed:
Fixed configtest failed

traceurgan committed:
Updated to master

marlenekoh committed:
refactoring timetable write and read file methods to storage

marlenekoh committed:
Update storagemanager with timetablestorage

marlenekoh committed:
refactoring

traceurgan committed:
Fix Tests

chenxing1992 committed:
Update Changes to remove addressbook

traceurgan committed:
minor edit

chenxing1992 committed:
Change personList to partner

chenxing1992 committed:
Merge branch 'removeAddressbook' of https://github.com/CS2103JAN2018-T09-B4/main into Main_Remove_Addressbook

traceurgan committed:
Edit calendar

marlenekoh committed:
Merged timetable into removeAddressBook

marlenekoh committed:
Update timetableutil

chenxing1992 committed:
Debug success to add appointment and delete appointment

traceurgan committed:
Minor edits

traceurgan committed:
Minor edit

traceurgan committed:
Merge branch 'Debug_Calendar' of https://github.com/CS2103JAN2018-T09-B4/main into removeAddressbook

# Conflicts:
#	src/main/java/seedu/address/logic/LogicManager.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/ui/BrowserPanel.java

Resolved conflicts

traceurgan committed:
Merge timetable branch

traceurgan committed:
handle NPE when person deleted

traceurgan committed:
Zac's storage modifications

chenxing1992 committed:
collate files

marlenekoh committed:
CSS edits to make status bar text and error text easier to see

chenxing1992 committed:
Update PPP

marlenekoh committed:
Fixed undo redo commands

traceurgan committed:
handle key press
tab gets commandbox focus

marlenekoh committed:
CSS changes

add border for selection
change scrollbar colour
change prompt text colour

traceurgan committed:
show selection border

marlenekoh committed:
Fixed bug where timetable does not update on click

marlenekoh committed:
Merge remote-tracking branch 'origin/UI' into UI

traceurgan committed:
Merge with marlenes PR

marlenekoh committed:
Fixed checkstyle

marlenekoh committed:
Merge pull request #154 from marlenekoh/UI

Replace address book names with NUSCouples in UI

traceurgan committed:
Merge branch 'removeAddressbook' of https://github.com/CS2103JAN2018-T09-B4/main into removeAddressbook

traceurgan committed:
minor edit

marlenekoh committed:
Merge branch 'master' into Created-PPP

marlenekoh committed:
temp

marlenekoh committed:
Merge remote-tracking branch 'CS2103JAN2018-T09-B4/master'

marlenekoh committed:
temp

marlenekoh committed:
Merge pull request #169 from chenxing1992/Created-PPP

Created PPP and Colate

traceurgan committed:
Tests

marlenekoh committed:
Add ppp & update docs

marlenekoh committed:
Merge branch 'master' into PPP

marlenekoh committed:
Merge pull request #171 from marlenekoh/PPP

Ppp

marlenekoh committed:
Merge remote-tracking branch 'CS2103JAN2018-T09-B4/removeAddressbook' into removeAddressbook

marlenekoh committed:
fix some tests

marlenekoh committed:
Merge branch 'removeAddressbook' into timetable

traceurgan committed:
Fixed test cases.

chenxing1992 committed:
Merge branch 'removeAddressbook' of https://github.com/CS2103JAN2018-T09-B4/main into Final_Fixes

chenxing1992 committed:
Fixing test #1

chenxing1992 committed:
fix bug #2

marlenekoh committed:
Fixed timetable not showing in jar

However, does not work in intellij due to path diff

marlenekoh committed:
Fixed undo indicate timetable changed

marlenekoh committed:
Merge branch 'removeAddressbook' into timetable

chenxing1992 committed:
bug fix #3

chenxing1992 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-T09-B4/main into Update_DG

marlenekoh committed:
checkstyle changes

traceurgan committed:
Add new class date to replace string date

traceurgan committed:
Show journal entry UI

marlenekoh committed:
Add function to let code be able to run rom intellij

traceurgan committed:
Can view journal
Changing journal entry automatically shows changes
on clear, shows calendar.

marlenekoh committed:
Changed select and deselect to tview and cview

Also some minor checkstyle fixes

marlenekoh committed:
Merge branch 'timetable' into removeAddressbook

marlenekoh committed:
Merge branch 'removeAddressbook' into timetable

marlenekoh committed:
checkstyle fixes

marlenekoh committed:
Merge branch 'timetable' into removeAddressbook

marlenekoh committed:
Update comments for ViewCalendarCommand and ViewTimetableCommand

also get rid of magic number acad year

traceurgan committed:
Tests

marlenekoh committed:
Update FileTimetableStorage constructor in StorageManagerTest

marlenekoh committed:
Merge branch 'master' into updateDocs

chenxing1992 committed:
add in view calendar feature

marlenekoh committed:
Merge branch 'timetable' into testbranch

marlenekoh committed:
Fixed addcommandtest and logicmanager test

renamed viewcalendar command and viewtimetable command to prevent confusion with calendar view command

marlenekoh committed:
Renamed viewtimetablecommand and viewcalendarcommand

to prevent confusion with calendarview command

marlenekoh committed:
checkstyle fixes

chenxing1992 committed:
created Add appointment feature

marlenekoh committed:
Fixed more tests

chenxing1992 committed:
Created Delete features

chenxing1992 committed:
Minor Updates

chenxing1992 committed:
move files

chenxing1992 committed:
Merge branch 'Update_DG' into Update-UG

chenxing1992 committed:
Update UG #1

chenxing1992 committed:
update UG and minor updates on DG and PPP too

chenxing1992 committed:
minor updates to PPP #1

chenxing1992 committed:
minor updates

chenxing1992 committed:
Update UI

chenxing1992 committed:
Merge branch 'testbranch' of https://github.com/CS2103JAN2018-T09-B4/main into Final_Fixes

traceurgan committed:
Fixed jsonuserprefsstoragetest

chenxing1992 committed:
rectify person and personbuilder

marlenekoh committed:
add more tests for timetable util and timetable

traceurgan committed:
Attempted fix

traceurgan committed:
Merge branch 'removeAddressbook' of https://github.com/CS2103JAN2018-T09-B4/main into removeAddressbook

marlenekoh committed:
temp

marlenekoh committed:
Merge branch 'removeAddressbook' into testbranch

traceurgan committed:
Tests

marlenekoh committed:
Add tests for showtimetablecommand and showcalendarcommand

marlenekoh committed:
Merge remote-tracking branch 'CS2103JAN2018-T09-B4/removeAddressbook' into testbranch

marlenekoh committed:
temp

traceurgan committed:
Fix tests

traceurgan committed:
One test left

marlenekoh committed:
update tests for filetimetablestorage

marlenekoh committed:
Merge branch 'removeAddressbook' into testbranch

marlenekoh committed:
Renamed timetable hashmaps so that it's more intuitive

traceurgan committed:
Add journal tests

traceurgan committed:
Ignore test case temporarily

traceurgan committed:
Merge branch 'master' into removeAddressbook

traceurgan committed:
Merge branch 'removeAddressbook' of https://github.com/traceurgan/main into removeAddressbook

traceurgan committed:
CS

marlenekoh committed:
Merge pull request #162 from traceurgan/removeAddressbook

Remove addressbook

traceurgan committed:
minor edit

marlenekoh committed:
Add tests for comparetimetablecommand 

+ make model getPartner() throw nullpointer

marlenekoh committed:
Merge branch 'removeAddressbook' into testbranch

traceurgan committed:
Added UI for view journal entry

marlenekoh committed:
remove getpartner throw null pointer exception

marlenekoh committed:
Merge branch 'testbranch' into timetable

marlenekoh committed:
Merge branch 'master' into timetable

chenxing1992 committed:
minor fixes

chenxing1992 committed:
test cases fix

traceurgan committed:
Test Fix

chenxing1992 committed:
Merge branch 'master' into Update_DG

chenxing1992 committed:
Merge branch 'master' into Update-UG

chenxing1992 committed:
Merge branch 'master' into Created-PPP

marlenekoh committed:
moved test resources from sanbox to view/resources

traceurgan committed:
remove tag

marlenekoh committed:
fixed test

traceurgan committed:
CS

marlenekoh committed:
Merge branch 'master' into updateDocs

traceurgan committed:
Added test for Date

marlenekoh committed:
changed TOC to show 3 levels

marlenekoh committed:
Merge pull request #175 from chenxing1992/Update_DG

Update DG

chenxing1992 committed:
minor updates

chenxing1992 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-T09-B4/main into Update-UG

chenxing1992 committed:
Merge branch 'Update-UG' of https://github.com/chenxing1992/main into Update-UG

marlenekoh committed:
Merge pull request #178 from marlenekoh/timetable

Add tests for Timetable feature

chenxing1992 committed:
Merge branch 'master' into Update-UG

marlenekoh committed:
Merge pull request #176 from chenxing1992/Update-UG

Update UG

chenxing1992 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-T09-B4/main into Created-PPP

marlenekoh committed:
Merge pull request #177 from chenxing1992/Created-PPP

Created ppp

marlenekoh committed:
Merge branch 'updateDocs'

marlenekoh committed:
edit userguide

marlenekoh committed:
Merge branch 'master' into updateDocs

chenxing1992 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-T09-B4/main into Final_Fixes

traceurgan committed:
Bug fix

traceurgan committed:
Minor bugfix

traceurgan committed:
Bug fix

traceurgan committed:
update docs

chenxing1992 committed:
Create Labels on list panel and update fixes

traceurgan committed:
Remove testcase

traceurgan committed:
CS

chenxing1992 committed:
remove unused import

traceurgan committed:
Merge branch 'master' into updatenewbranch

marlenekoh committed:
add images

marlenekoh committed:
Merge branch 'master' into updateDocs

marlenekoh committed:
Add pics to user guide

marlenekoh committed:
Update user guide pictures

chenxing1992 committed:
Update image for googleAPI sequence Diagram

chenxing1992 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-T09-B4/main into Update-UG

marlenekoh committed:
Merge pull request #182 from marlenekoh/updateDocs

Add pictures to user guide

chenxing1992 committed:
update UG

chenxing1992 committed:
Updated merge conflict

chenxing1992 committed:
Merge branch 'master' into Final_Fixes

chenxing1992 committed:
Update UG

chenxing1992 committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-T09-B4/main into Final_Fixes

chenxing1992 committed:
Merge branch 'Final_Fixes' of https://github.com/chenxing1992/main into Final_Fixes

chenxing1992 committed:
insert new sequence diagram

traceurgan committed:
Update docs

traceurgan committed:
Merge branch 'updatenewbranch' of https://github.com/traceurgan/main into update

e0003604 committed:
Updated tags for user guide

e0003604 committed:
Added PPP

e0003604 committed:
Added PPP

marlenekoh committed:
Update DG

marlenekoh committed:
Merge pull request #183 from chenxing1992/Update-UG

Update UG

marlenekoh committed:
Merge branch 'master' into updateDocs

traceurgan committed:
Merge branch 'master' into updatenewbranch

e0003604 committed:
Merge branch 'master' into master

marlenekoh committed:
Merge pull request #184 from marlenekoh/updateDocs

Update DG

traceurgan committed:
PPP

chenxing1992 committed:
Merge branch 'master' into Final_Fixes

marlenekoh committed:
update DG

marlenekoh committed:
Merge remote-tracking branch 'origin/updateDocs' into updateDocs

traceurgan committed:
merge

marlenekoh committed:
Merge branch 'master' into updateDocs

marlenekoh committed:
Merge pull request #186 from marlenekoh/updateDocs

Update Developer guide

marlenekoh committed:
Merge branch 'master' into Final_Fixes

marlenekoh committed:
Merge pull request #180 from chenxing1992/Final_Fixes

Create Labels on panel list

marlenekoh committed:
Merge branch 'master' into master

marlenekoh committed:
Merge pull request #185 from HEARTOFAL1ON/master

Added PPP

traceurgan committed:
Resolve merge conflict

chenxing1992 committed:
create pdf docs

marlenekoh committed:
Merge pull request #187 from traceurgan/testmerge

Update docs

chenxing1992 committed:
Files collated

marlenekoh committed:
add more implementation notes to developer guide

marlenekoh committed:
Merge branch 'master' into updateDocs

marlenekoh committed:
Merge pull request #189 from marlenekoh/updateDocs

add more implementation notes to developer guide

marlenekoh committed:
Merge branch 'master' into Collate

marlenekoh committed:
Merge pull request #188 from chenxing1992/Collate

Collate

traceurgan committed:
update collate tests

T09B4 committed:
Merge pull request #190 from traceurgan/collate

update collate

marlenekoh committed:
add collate unused for securityutil

marlenekoh committed:
remove pdf files to pass travis
